Source/Scene/ColorConditional.js

```diff
@@ -0,0 +1,130 @@
+/*global define*/
+define([
+       './Expression',
+       '../Core/clone',
+       '../Core/Color',
+       '../Core/defaultValue',
+       '../Core/defined',
+       '../Core/defineProperties'
+    ], function(
+        Expression,
+        clone,
+        Color,
+        defaultValue,
+        defined,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * DOC_TBA
+     * <p>
+     * Do not construct this directly; instead use {@link Cesium3DTileStyle}.
+     * </p>
+     */
+    function ColorConditional(styleEngine, jsonExpression) {
+        this._styleEngine = styleEngine;
+        this._conditional = clone(jsonExpression.conditional, true);
+        this._default = jsonExpression.default;
+
+        this._runtimeConditional = undefined;
+        this._runtimeDefault = undefined;
+
+        setRuntime(this);
+        setRuntimeDefault(this);
+    }
+
+    defineProperties(ColorConditional.prototype, {
+        /**
+         * DOC_TBA
+         */
+        conditional : {
+            get : function() {
+                return this._conditional;
+            },
+            set : function(value) {
+                this._conditional = clone(value, true);
+                setRuntime(this);
+                this._styleEngine.makeDirty();
+            }
+        },
+
+        /**
+         * DOC_TBA
+         */
+        default : {
+            get : function() {
+                return this._default;
+            },
+            set : function(value) {
+                if (value !== this._default) {
+                    this._default = value;
+                    setRuntimeDefault(this);
+                    this._styleEngine.makeDirty();
+                }
+            }
+        }
+    });
+
+    function setRuntime(expression) {
+        var runtimeConditional = [];
+        var conditional = expression._conditional;
+        for (var cond in conditional) {
+            if (conditional.hasOwnProperty(cond)) {
+                var colorExpression = conditional[cond];
+                runtimeConditional.push({
```

Comment:
Create a constructor function (local to this file) for this implicit type.  It will help with performance.

By the way, for performance tips, check this out when you have time: https://www.youtube.com/watch?v=XAqIpGU8ZZk